dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -0,0 +1,358 @@
+                &m_notificationComActivatorRegistration));
+        }
+    }
+
+    void AppNotificationManager::Unregister()
+    {
+        auto lock{ m_lock.lock_exclusive() };
+        THROW_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), !m_notificationComActivatorRegistration, "Not Registered for App Notifications!");
+        m_notificationComActivatorRegistration.reset();
+    }
+
+    void AppNotificationManager::UnregisterAll()
+    {
+        Unregister();
+
+        // Remove any Registrations from the Long Running Process that are necessary for Cloud toasts
+        if (!PushNotificationHelpers::IsPackagedAppScenario())
+        {
+            wil::unique_cotaskmem_string processName;
+            THROW_IF_FAILED(GetCurrentProcessPath(processName));
+            auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+            THROW_IF_FAILED(notificationPlatform->RemoveToastRegistrationMapping(processName.get()));
+        }
+
+        if (!AppModel::Identity::IsPackagedProcess())
+        {
+            std::wstring notificationId{ RetrieveNotificationAppId() };
+            std::wstring storedComActivatorString;
+            THROW_IF_FAILED(GetActivatorGuid(storedComActivatorString));
+            UnRegisterComServer(storedComActivatorString);
+            
+            UnRegisterNotificationAppIdentifierFromRegistry();
+            THROW_IF_FAILED(PushNotifications_UnregisterFullTrustApplication(notificationId.c_str()));
+        }
+    }
+
+    winrt::event_token AppNotificationManager::NotificationInvoked(winrt::Windows::Foundation::TypedEventHandler<winrt::Microsoft::Windows::AppNotifications::AppNotificationManager, winrt::Microsoft::Windows::AppNotifications::AppNotificationActivatedEventArgs> const& handler)
+    {
+        auto lock{ m_lock.lock_exclusive() };
+        THROW_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), m_notificationComActivatorRegistration, "Must register event handlers before calling Register()");
+        return m_notificationHandlers.add(handler);
+    }
+
+    void AppNotificationManager::NotificationInvoked(winrt::event_token const& token) noexcept
+    {
+        auto lock{ m_lock.lock_exclusive() };
+        m_notificationHandlers.remove(token);
+    }
+
+    HRESULT __stdcall AppNotificationManager::Activate(
+        LPCWSTR appUserModelId,
+        LPCWSTR invokedArgs,
+        [[maybe_unused]] NOTIFICATION_USER_INPUT_DATA const* data,
+        [[maybe_unused]] ULONG dataCount) noexcept try
+    {
+
+        winrt::IMap<winrt::hstring, winrt::hstring> userInput{ winrt::single_threaded_map<winrt::hstring, winrt::hstring>() };
+        for (unsigned long i = 0; i < dataCount; i++)
+        {
+            userInput.Insert(data[i].Key, data[i].Value);
+        }
+
+        winrt::AppNotificationActivatedEventArgs activatedEventArgs = winrt::make<winrt::Microsoft::Windows::AppNotifications::implementation::AppNotificationActivatedEventArgs>(invokedArgs, userInput);
```

Comment:
>winrt::Microsoft::Windows::AppNotifications::implementation:: [](http://example.com/codeflow?start=82&length=61)

Within the namespace, you can say `implementation::AppNotificationActivatedEventArgs` to get the right type.